app/javascript/packs/base.js

```diff
@@ -4,6 +4,9 @@ import {
   getInstantClick,
   initializeTouchDevice,
 } from '../topNavigation/utilities';
+import { getFocusTrapToggle } from '../utilities/getFocusTrapToggle';
+
+window.getFocusTrapToggle = getFocusTrapToggle;
```

Comment:
refactor (blocking): load the getFocusTrapToggle asynchronously. Modals are on demand by nature and we're also trying to avoid the extra payload that the focus-trap package provides. The best way to handle this would be via a dynamic import. (see the mention about this in https://github.com/forem/forem/pull/12366#discussion_r565454148)

```suggestion
window.getFocusTrapToggle = async (selector) => {
  const { getFocusTrapToggle } = await import('@utilities/getFocusTrapToggle');

  return getFocusTrapToggle(selector);
};
```

And at the call site, you could do the following:

```js
function showModal(context) {
  window
    .getFocusTrapToggle('#global-signup-modal')
    .then((toggleSignupModal) => {
      toggleSignupModal();
      document.body.classList.add('modal-open');

      if (document.getElementById('global-signup-modal')) {
        document.getElementsByClassName(
          'authentication-modal__close-btn',
        )[0].onclick = () => {
          toggleSignupModal();
          document.body.classList.remove('modal-open');
        };
      }
    });
}
```

Also, the `getFocusTrapToggle` seems to have too many responsibilities. At first I thought it was just to enable the focus trap in the modal, but after testing it out. It's really a open and close modal function that happens to also enable/disable the focus trap. I thought about this a bit and I think we can actually load the Preact Modal component instead in all our cases. I need to sleep on it, but I think it should be doable, even for the admin area that uses Stimulus.  Let's sync up tomorrow to discuss.

Reply:
Yeah it would be good to chat this over. If we can use the Preact modal then that'd be great - to be honest I avoided doing that right now because of both stimulus stuff, and the non-Preact modals toggling visibility in a different way. It just seemed like a bigger refactor to change that here. Also the listings modal would require some more refactoring to make it work inside the Preact modal just since it has some different styling.

Reply:
Applied the dynamic import change for now. Thinking about it, `getFocusTrapToggle` is poor naming and `getModalToggle` makes more sense with the responsibilities/usages it has (since opening/closing a modal _should_ create/disable a focus trap) 🙃